src/LoanActionsHistory.js

```diff
@@ -134,23 +130,31 @@ class LoanActionsHistory extends React.Component {
       nonRenewedLoansModalOpen: false,
       changeDueDateDialogOpen: false,
       requestsCount: {},
-      loanPolicyName: '',
       feesFines: {},
       patronBlockedModal: false,
     };
   }
 
   componentDidMount() {
-    this.getOpenRequestsCount();
-    this.getLoanPolicyName();
     this.getLoanActions();
     this.getFeeFine();
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.loan.itemId !== prevProps.loan.itemId) {
-      this.getOpenRequestsCount();
-      this.getLoanPolicyName();
+    const {
+      loan: {
+        renewalCount
+      }
+    } = this.props;
+
+    const {
+      loan: {
+        renewalCount: prevRenewalCount
+      }
+    } = prevProps;
+
+    if (prevRenewalCount && renewalCount && renewalCount !== prevRenewalCount) {
```

Comment:
I would suggest moving that logic inside `componentDidUpdate` to the separate method and call it inside `componentDidUpdate`.

Additionally it might be good idea to assign expression bellow to the variable.
`prevRenewalCount && renewalCount && renewalCount !== prevRenewalCount`